Oneirocom/Magick
# Improve node menu

## What Changed:

- Completely redid our node categories to make use of subcategories.
- Improved search visibility by removing subcategory nesting on search
- added support for a common flag on components to add them to a common category 
- Added a common category to the top of the menu for easy access

## How to test:

Explore the node menu.  Search for nodes.  Check if all node categories make sense.

## Additional information:


https://github.com/Oneirocom/Magick/assets/2397603/984bfca4-c362-45f4-baf8-8268bb3d7e46
